pkg/secretless/plugin/sharedobj/available_plugins_test.go

```diff
@@ -1,60 +1,166 @@
 package sharedobj
 
 import (
+	"errors"
 	"testing"
 
-	"github.com/cyberark/secretless-broker/pkg/secretless/plugin"
 	"github.com/stretchr/testify/assert"
 
-	"github.com/cyberark/secretless-broker/pkg/secretless/plugin/connector/http"
-	"github.com/cyberark/secretless-broker/pkg/secretless/plugin/connector/tcp"
+	log_api "github.com/cyberark/secretless-broker/pkg/secretless/log"
+	"github.com/cyberark/secretless-broker/pkg/secretless/plugin"
+	"github.com/cyberark/secretless-broker/pkg/secretless/plugin/sharedobj/mock"
 )
 
-type mockHTTPPlugin struct{ http.Plugin }
-type mockTCPPlugin struct{ tcp.Plugin }
-
-// TODO: Test AllAvailablePluginsWithOptions
-
-// These need to sit outside of the func since we are comparing them in the
-// tests
-var mockHTTPPlugins = map[string]http.Plugin{
-	"one": mockHTTPPlugin{},
-	"two": mockHTTPPlugin{},
+func TestCompatiblePluginVersion(t *testing.T) {
+	assert.Equal(t, "0.1.0", CompatiblePluginAPIVersion)
```

Comment:
It's unclear to me what the value of this test is.  If you forgot to update this on a version change, the test would continue to pass.  It would only be valuable if you happened to remember to update the test but forgot to update the implementation.  It seems like rather than a test we want an item in a checklist somewhere that would remind us to update the actual impl value when appropriate.

Reply:
Maybe the better test would be a regex that asserts that we are in semVer format?